src/legacy/state/orders/updaters/SpotPricesUpdater.ts

```diff
@@ -25,26 +25,28 @@ type MarketRecord = Record<
   }
 >
 
-function useMarkets(chainId?: SupportedChainId): MarketRecord {
+function useMarkets(chainId: SupportedChainId): MarketRecord {
   const pending = useCombinedPendingOrders({ chainId })
 
   return useSafeMemo(() => {
     return pending.reduce<Record<string, { chainId: number; inputCurrency: Token; outputCurrency: Token }>>(
       (acc, order) => {
-        if (chainId) {
-          // Aggregating pending orders per market. No need to query multiple times same market
-          const { marketInverted, marketKey } = getCanonicalMarketChainKey(chainId, order.sellToken, order.buyToken)
+        // Query spot prices only for Limit orders
+        if (order.class !== OrderClass.LIMIT) return acc
```

Comment:
We still show the market price and estimated market price on the TWAP orders table, no?